pkgs/tools/typesetting/tex/texlive/combine.nix

```diff
@@ -35,16 +35,14 @@ let
 
   sortedUniqueStrings = list: lib.sort (a: b: a < b) (lib.unique list);
 
-  mkUniqueOutPaths = pkgs: lib.unique
-    (map (p: p.outPath) (builtins.filter lib.isDerivation pkgs));
-
 in (buildEnv {
   name = "texlive-${extraName}-${bin.texliveYear}${extraVersion}";
 
   extraPrefix = "/share/texmf";
 
   ignoreCollisions = false;
-  paths = pkgList.nonbin;
+  # remove irrelevant source of derivation changes by sorting
+  paths = lib.sort (a: b: a.outPath < b.outPath) (pkgList.nonbin);
```

Comment:
Is this for use in `nix-diff`?

Reply:
This is to avoid a recompile if dependencies appear in a different order, for instance if a user calls `texlive.combine` in a slightly different way. The output does not depend on the order, so avoiding a (slow) recompile is nice. I couldn't measure an effect on the evaluation time.